rpcserverhelp.go

```diff
@@ -839,7 +839,9 @@ var helpDescsEnUS = map[string]string{
 	"estimatesmartfee--synopsis":     "Returns the estimated fee using the historical fee data in dcr/kb.",
 	"estimatesmartfee-confirmations": "Estimate the fee rate a transaction requires so that it is mined in up to this number of blocks.",
 	"estimatesmartfee-mode":          "The only supported mode for the moment is 'conservative'.",
-	"estimatesmartfee--result0":      "Estimated fee rate (in DCR/KB).",
+	"estimatesmartfeeresult-feerate": "The Estimated fee rate (in DCR/KB).",
+	"estimatesmartfeeresult-errors":  "Unused.",
+	"estimatesmartfeeresult-blocks":  "The block number where the estimate was found.",
```

Comment:
I this the same as `c.Confirmations`?

Reply:
Yeah.  I don't think the description is really that accurate.

Reply:
On btc core which was the basis for this, this returns either the passed conf target, 2 (if the requested conf target == 1) or the max usable usable conf target for estimates. The source for this statement is the follwoing:

- The `estimatesmartfee` rpc endpoint determines the `blocks` return value [here](https://github.com/bitcoin/bitcoin/blob/65a54d684fcfa5b2cb15b09fd9be9acb56bb5f79/src/rpc/mining.cpp#L1052-L1060)
- Notice that the `blocks` return value comes back from `feeCalc->returnedTarget`
- `returnedTarget` is only ever set [here](https://github.com/bitcoin/bitcoin/blob/65a54d684fcfa5b2cb15b09fd9be9acb56bb5f79/src/policy/fees.cpp#L789-L809)
- Notice it defaults to the `confTarget` argument and only adjusts on the `== 1` and the `>= maxUsableEstimate` cases

So all in all, it seems like a somewhat useless return value on btc core.

I think the _intention_ of using it to signal the smallest bucket that still fulfills the required parameter is reasonable (so, if you request the fee for a confTarget of `10` but the estimator finds that at a confTarget of `5` you already pay the same fees  as for `10` it would return `blocks == 5`) but that's not how core behaves _in practice_ today.

The question is then: do we want to follow the same convention as core, do we want to drop this altogether (basically the current implementation), change it to the desired intention I wrote above or something else?

Reply:
I referred to this [issue](https://github.com/btcsuite/btcd/issues/1146#issuecomment-435202838) mostly to discern what `blocks` was about. Also from the core implementation matheusd shared `feeCalc->returnedTarget` is either the passed in `confTarget` or `MaxUsableEstimate`. From dcrd's `estimatesmartfee` implementation I didn't notice any limiter identical to `MaxUsableEstimate` so it made sense to set `c.Confirmations` as `blocks`.

Reply:
Maybe just me, but the name of the argument is confusing. Wouldn't it make more sense as `blocks`? Does a transaction get "confirmations" before it is mined/included in a block? Isn't the argument the number of blocks until the transaction will be mined and start getting confirmations? Am I misunderstanding the fundamentals?

Reply:
I just realized this in the fees readme:
```
- Ensuring transactions are mined within a target _confirmation range_
  (expressed in blocks);
```
So I guess this is known terminology.